### PR TITLE
Upgrade lisk-js version and adapt source code - Partially addresses #461

### LINKED ISSUE #461: Update Lisk-Elements dependency to 1.0.0
- Need to delete BETA_NET options.

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?
Needed to upgrade to 1.0.0 branch

### How did I fix it?
Upgrade to lisk-js 1.0.0 branch and adapt accordingly to use APIClient.

### How to test it?
`npm test`
and try get, set command and transactions, signature broadcasting.
### Review checklist

* The PR is a partial solution to #461 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Comment by willclarktech** [comment]
Updated to say this is a partial solution to #461

File: package.json [hunk 1]
```diff
@@ -55,7 +55,7 @@
 		"bip39": "=2.4.0",
 		"chalk": "=2.3.0",
 		"cli-table2": "=0.2.0",
-		"lisk-js": "LiskHQ/lisk-js#1.0.0-dev-lisky",
+		"lisk-js": "LiskHQ/lisk-js#b18cf68d9e17ad1849ca22c96932c82235f22599",
```
Comment (willclarktech) [inline package.json]: We have a branch on `lisk-js` so that we don't have to constantly update commit hashes: we can just update that branch to whatever we need.

File: src/commands/get.js [hunk 2]
```diff
@@ -15,7 +15,12 @@
  */
 import { COMMAND_TYPES } from '../utils/constants';
 import { ValidationError } from '../utils/error';
-import { createCommand, deAlias, processQueryResult } from '../utils/helpers';
+import {
+	createCommand,
+	deAlias,
+	processQueryResult,
+	checkBoolean,
```
Comment (willclarktech) [inline src/commands/get.js]: Should this be part of this PR?

File: src/commands/get.js [hunk 3]
```diff
@@ -35,9 +40,13 @@ export const actionCreator = () => async ({
 		throw new ValidationError('Unsupported type.');
 	}
 
-	return query.handlers[deAlias(type)](input, { testnet }).then(
-		processQueryResult(type),
-	);
+	const overrideTestnet = checkBoolean(testnet)
```
Comment (willclarktech) [inline src/commands/get.js]: This sounds like a function name. Prefer `testnetOverrideValue` or something similar.

File: src/utils/helpers.js [hunk 4]
```diff
@@ -86,10 +87,12 @@ export const validateAmount = amount => {
 	return true;
 };
 
+export const checkBoolean = value => ['true', 'false'].includes(value);
```
Comment (willclarktech) [inline src/utils/helpers.js]: Unit tests?

File: src/utils/constants.js [hunk 5]
```diff
@@ -29,4 +29,12 @@ export const SINGULARS = {
 	transactions: 'transaction',
 };
 
+export const PLURALS = {
```
Comment (willclarktech) [inline src/utils/constants.js]: Test?

File: src/utils/api.js [hunk 6]
```diff
@@ -16,6 +16,28 @@
...
-const { api: LiskAPI } = liskJS;
+const { APIClient, constants } = liskJS;
 
-export default new LiskAPI(config.liskJS);
+const defaultClient =
+	config.liskJS.testnet === true
+		? APIClient.createTestnetAPIClient(config.liskJS)
+		: APIClient.createMainnetAPIClient(config.liskJS);
+
+export const setAPIClientOptions = (providedOptions = {}) => {
```
Comment (willclarktech) [inline src/utils/api.js]: Unit tests?

File: src/commands/get.js [hunk 7]
```diff
@@ -35,9 +40,13 @@ export const actionCreator = () => async ({
...
+		? testnet === 'true'
+		: undefined;
+
+	return query.handlers[deAlias(type)](input, {
```
Comment (willclarktech) [inline src/commands/get.js]: What about singular/plurals?

File: src/utils/constants.js [hunk 8]
```diff
@@ -14,19 +14,26 @@
...
+	'transactions',
 ];
 
-export const SINGULARS = {
-	accounts: 'account',
-	addresses: 'address',
-	blocks: 'block',
-	delegates: 'delegate',
-	transactions: 'transaction',
+export const PLURALS = {
```
Comment (willclarktech) [inline src/utils/constants.js]: We could use some sanity check tests for this file.

File: src/utils/query.js [hunk 9]
```diff
@@ -13,70 +13,20 @@
...
-	getDelegate(input, options) {
-		return this.sendRequest('delegates/get', { username: input }, options);
-	}
-}
-
-export default new Query();
+import getAPIClient from './api';
+
+export default (endpoint, parameters, { testnet } = {}) =>
+	// prettier-ignore
```
Comment (willclarktech) [inline src/utils/query.js]: Why is this necessary?

File: src/utils/query.js [hunk 10]
```diff
@@ -13,70 +13,20 @@
...
+	getAPIClient(testnet)[endpoint].get(parameters)
+		.then(res => {
+			if (res.data) {
+				if (Array.isArray(res.data)) {
+					if (res.data.length === 0) {
+						throw new Error('Data was not found with specified parameters.');
```
Comment (willclarktech) [inline src/utils/query.js]: This is a pretty obscure error message. Slightly better would be `No results were found with the specified parameters.` Can we customise it for the endpoint and/or parameters though?

File: src/commands/list.js [hunk 11]
```diff
@@ -31,21 +31,23 @@ export const actionCreator = () => async ({
...
-		query.handlers[deAlias(singularType)](input, { testnet }),
-	);
+	const endpoint = deAlias(pluralType);
 
-	return Promise.all(queries).then(results =>
-		results.map(processQueryResult(singularType)),
-	);
+	const queries = inputs.map(input => {
+		const req = {
+			limit: 1,
```
Comment (willclarktech) [inline src/commands/list.js]: Why do we need this for `list`?

File: src/utils/api.js [hunk 12]
```diff
@@ -13,9 +13,17 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-import liskJS from 'lisk-js';
+import lisk from 'lisk-js';
```
Comment (willclarktech) [inline src/utils/api.js]: Did we consider `liskElements` or just `elements`? @Tosch110 what are your thoughts?

Comment (shuse2) [reply src/utils/api.js]: yes I was thinking to change it to `elements`, but not officially changed name.
so, I settled in kinda middle...

File: src/commands/list.js (hunk 11 above)
Comment (shuse2) [reply src/commands/list.js]: it's because there is no way to specify array or multiple ids on the Core API.
Therefore we need multiple requests

File: src/utils/query.js (hunk 9 above)
Comment (shuse2) [reply src/utils/query.js]: It put the weird line break in the middle...

File: src/utils/constants.js (hunk 8 above)
Comment (shuse2) [reply src/utils/constants.js]: I will create the different issue for that!

File: src/utils/query.js (hunk 10 above)
Comment (willclarktech) [reply src/utils/query.js]: I mostly object to the word `data`.

Comment (shuse2) [reply src/utils/query.js]: maybe instead of data, we use endpoint?
like
```
lisky> get delegate 123
╔═══════════════════════════════════════════════════════════════════╗
║ error                                                             ║
╟───────────────────────────────────────────────────────────────────╢
║ Could not get: delegates was not found with specified parameters. ║
╚═══════════════════════════════════════════════════════════════════╝
```

Comment (willclarktech) [reply src/utils/query.js]: `Could not get: delegates not found using specified endpoints.`?

Comment (willclarktech) [reply src/utils/query.js]: Or `no delegates found using specified endpoints`? They're reliably plural, right?

Comment (shuse2) [reply src/utils/query.js]: Most of them are plural except `node` endpoint, which we don't have now on commander,
so referring as plural should be ok.
But I think the error message should be refer to parameters or options? rather than end points I think.
- user don't think about endpoint at this point
- event with the correct endpoint, it's possible to get no result depending on the paramter

Comment (willclarktech) [reply src/utils/query.js]: Sorry, I meant `parameters`, not `endpoints`.

File: src/utils/api.js (hunk 12 above)
Comment (willclarktech) [reply src/utils/api.js]: As discussed, we'll leave it as it is for now.

File: test/steps/queries/3_then.js [hunk 13]
```diff
@@ -13,26 +13,30 @@
...
+	return expect(returnValue).to.eventually.be.an('object');
 }
 
-export function theQueryInstanceShouldHaveAHandlerFor() {
-	const { queryInstance } = this.test.ctx;
-	const item = getFirstQuotedString(this.test.title);
-	return expect(queryInstance.handlers)
-		.to.have.property(item)
-		.be.a('function');
+export function itShouldResolveToAnEmptyObject() {
```
Comment (willclarktech) [inline test/steps/queries/3_then.js]: These two steps are pretty generic and probably belong in the `general` module.

File: test/steps/queries/3_then.js [hunk 14]
```diff
@@ -13,26 +13,30 @@
...
+	const { returnValue } = this.test.ctx;
+	return expect(returnValue).to.eventually.be.eql({});
+}
+
+export function itShouldResolveToTheDataKeyOfTheResponse() {
```
Comment (willclarktech) [inline test/steps/queries/3_then.js]: It's not the key, it's the data.

File: test/steps/queries/3_then.js [hunk 15]
```diff
@@ -13,26 +13,30 @@
...
+	const { returnValue, sendRequestResult } = this.test.ctx;
+	return expect(returnValue).to.eventually.equal(sendRequestResult.data);
+}
+
+export function itShouldResolveToTheFirstElementOfDataKeyOfTheResponse() {
```
Comment (willclarktech) [inline test/steps/queries/3_then.js]: As above.

File: test/steps/queries/2_when.js [hunk 16]
```diff
@@ -14,31 +14,64 @@
...
+	liskAPIInstance[endpoint].get.resolves(sendRequestResult);
+	this.test.ctx.sendRequestResult = sendRequestResult;
+
+	const returnValue = query(endpoint, parameters, options);
+	this.test.ctx.returnValue = returnValue;
+
+	return returnValue.catch(e => e);
+}
+
+export function theQueryInstanceSendsARequestAndTheLiskAPIInstanceResolvesWithASuccessfulResponseOfEmptyArray() {
```
Comment (willclarktech) [inline test/steps/queries/2_when.js]: `theQueryInstanceSendsARequestAndTheLiskAPIInstanceResolvesWithASuccessfulEmptyArrayDataResponse` would be more consistent with the step above.

File: test/steps/config/1_given.js [hunk 17]
```diff
@@ -45,6 +45,14 @@ export function aConfigWithJsonSetTo() {
 	this.test.ctx.config = config;
 }
 
+export function aConfigWithLiskJSTestnetSetTo() {
+	const bool = getFirstBoolean(this.test.parent.title);
```
Comment (willclarktech) [inline test/steps/config/1_given.js]: Prefer `const testnet` directly

File: test/specs/utils/query.js [hunk 18]
```diff
@@ -18,208 +18,108 @@ import * as given from '../../steps/1_given';
...
-						);
-					},
-				);
-				When(
-					'the query instance sends a request and the lisk API instance resolves with a failed response',
-					when.theQueryInstanceSendsARequestAndTheLiskAPIInstanceResolvesWithAFailedResponse,
+		Given('an endpoint "accounts"', given.anEndpoint, () => {
+			Given('a parameters object', given.aParametersObject, () => {
+				Given(
+					'the parameters object has key set to key="address" value="new-address"',
```
Comment (willclarktech) [inline test/specs/utils/query.js]: `the parameters object has key "address" set to "new-address"`
